pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -151,7 +151,15 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environme
 
 	for _, service := range push.Services(proxy) {
 		destRule := push.DestinationRule(proxy, service)
+		mtlsReady := push.GetMTLSReady(service.Hostname)
```

Comment:
why not just `Service.MTLSReady`?

Reply:
+1

We should avoid iterating in CDS code.

Reply:
Is this possible that different port can have different group of endpoints. In other words, can mTLSReady state changed for different ports?